tests/kernel/threads/thread_stack/src/main.c

```diff
@@ -414,10 +414,76 @@ void test_stack_buffer(void)
 		       STEST_STACKSIZE, false);
 }
 
+void no_op_entry(void *p1, void *p2, void *p3)
+{
+
+	printk("hi! bye!\n");
+
+#ifdef CONFIG_DYNAMIC_OBJECTS
+	/* Allocate a dynamic kernel object, which gets freed on thread
+	 * cleanup since this thread has the only reference.
+	 */
+	struct k_sem *dyn_sem = k_object_alloc(K_OBJ_SEM);
+	k_sem_init(dyn_sem, 1, 1);
+	printk("allocated semaphore %p\n", dyn_sem);
+#endif
+	/* thread self-aborts, triggering idle thread cleanup */
+}
+
+/**
+ * @brief Show that the idle thread stack size is correct
+ *
+ * The idle thread has to occasionally clean up self-exiting threads.
+ * Exercise this and show that we didn't overflow, reporting out stack
+ * usage.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_idle_stack(void)
```

Comment:
Add this new test case as a separate commit?

Reply:
will do